Source/WebKit/UIProcess/BrowsingContextGroup.cpp

```diff
@@ -68,30 +68,30 @@ void BrowsingContextGroup::sharedProcessForSite(WebsiteDataStore& websiteDataSto
         if (RefPtr frameProcess = protectedThis->m_sharedProcess.get())
             return completionHandler(frameProcess.get());
 
-        Ref process = pageConfiguration->protectedProcessPool()->processForSite(websiteDataStore.get(), site, lockdownMode, enhancedSecurity, pageConfiguration.get(), ProcessSwapDisposition::Other);
-        Ref frameProcess = FrameProcess::create(process, protectedThis, std::nullopt, preferences, InjectBrowsingContextIntoProcess::Yes);
+        Ref process = pageConfiguration->protectedProcessPool()->processForSite(websiteDataStore.get(), WebProcessPool::IsSharedProcess::Yes, site, mainFrameSite, lockdownMode, enhancedSecurity, pageConfiguration.get(), ProcessSwapDisposition::Other);
+        Ref frameProcess = FrameProcess::create(process, protectedThis, std::nullopt, mainFrameSite, preferences, InjectBrowsingContextIntoProcess::Yes);
         protectedThis->m_sharedProcess = frameProcess.ptr();
         completionHandler(frameProcess.ptr());
     });
 }
 
-Ref<FrameProcess> BrowsingContextGroup::ensureProcessForSite(const Site& site, WebProcessProxy& process, const WebPreferences& preferences, InjectBrowsingContextIntoProcess injectBrowsingContextIntoProcess)
+Ref<FrameProcess> BrowsingContextGroup::ensureProcessForSite(const Site& site, const Site& mainFrameSite, WebProcessProxy& process, const WebPreferences& preferences, InjectBrowsingContextIntoProcess injectBrowsingContextIntoProcess)
 {
     if (preferences.siteIsolationEnabled()) {
         if (m_sharedProcess && m_sharedProcessSites.contains(site)) {
             ASSERT(&m_sharedProcess->process() == &process);
             return *m_sharedProcess;
         }
-        if (RefPtr existingProcess = processForSite(site)) {
+        if (RefPtr existingProcess = processForSite(site, mainFrameSite)) {
             if (existingProcess->process().coreProcessIdentifier() == process.coreProcessIdentifier())
                 return existingProcess.releaseNonNull();
         }
     }
 
-    return FrameProcess::create(process, *this, site, preferences, injectBrowsingContextIntoProcess);
+    return FrameProcess::create(process, *this, site, mainFrameSite, preferences, injectBrowsingContextIntoProcess);
 }
 
-FrameProcess* BrowsingContextGroup::processForSite(const Site& site)
+FrameProcess* BrowsingContextGroup::processForSite(const Site& site, const Site& /* mainFrameSite */)
```

Comment:
Seems like mainFrameSite does not need to be a parameter here.

Reply:
oh yeah, will remove.